Text/AbbyyOCR/AbbyyOCR.cs

```diff
@@ -0,0 +1,257 @@
+
+        private static int _retryCount = 3;
+        private static int _delayBetweenRetriesInSeconds = 3;
+        private static string _httpClientName = "OCR_HTTP_CLIENT";
+
+		private static ConcurrentBag<WebApiResponseRecord> bag = new ConcurrentBag<WebApiResponseRecord>();
+
+
+		private static readonly AuthInfo AuthInfo = new AuthInfo
+        {
+            Host = ServiceUrl,
+            ApplicationId = ApplicationId,
+            Password = Password
+        };
+
+        private static ServiceProvider _serviceProvider;
+        private static HttpClient _httpClient;
+
+        [FunctionName("AbbyyOCR")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequest req,
+            ILogger log,
+            ExecutionContext executionContext)
+
+        {
+            log.LogInformation("ABBYY OCR Custom Skill: C# HTTP trigger function processed a request.");
+
+            string skillName = executionContext.FunctionName;
+            IEnumerable<WebApiRequestRecord> requestRecords = WebApiSkillHelpers.GetRequestRecords(req);
+            if (requestRecords == null)
+            {
+                return new BadRequestObjectResult($"{skillName} - Invalid request record array.");
+            }
+
+			// Create the response record
+			WebApiSkillResponse response = new WebApiSkillResponse();
+
+			using (var ocrClient = GetOcrClientWithRetryPolicy())
+			{
+				List<Task> TaskList = new List<Task>();
+				int idx = 0;
+				foreach (var record in requestRecords)
+				{
+					TaskList.Add(ProcessRecord(record, ocrClient, idx));
+					idx += 1;
+				}
+				Task.WaitAll(TaskList.ToArray());
+
+				// Apply all the records from the concurrent bag
+				foreach (var record in bag)
+					response.Values.Add(record);
+			}
+
+            return new OkObjectResult(response);
+
+        }
+
+		public static async Task ProcessRecord(WebApiRequestRecord record, IOcrClient ocrClient, int idx)
+		{
+			WebApiResponseRecord waRecord = new WebApiResponseRecord();
+
+			record.Data.TryGetValue("formUrl", out object imgFile);
+			record.Data.TryGetValue("formSasToken", out object sasToken);
+			string imgFileWithSaS = imgFile.ToString() + sasToken.ToString();
+			string fileType = imgFile.ToString().Substring(imgFile.ToString().LastIndexOf("."));
+			string localFile = Path.GetTempPath() + "\\" + "temp_" + idx.ToString() + fileType;
+			using (var client = new WebClient())
+			{
+				client.DownloadFile(imgFileWithSaS, localFile);
+			}
+
+			// Process image 
+			// You could also call ProcessDocumentAsync or any other processing method declared below
+			var resultUrls = await ProcessImageAsync(ocrClient, localFile);
+
+			//Get results - the first doc is a docx, second is a text file
+			using (var client = new WebClient())
+			{
+				waRecord.Data.Add("content", client.DownloadString(resultUrls[1].ToString()));
```

Comment:
Can we do some bounds validation here on `resultUrls`?

Reply:
Sorry, I don't know what it means to do bounds validation?

Reply:
Oh, I get it.  Sorry.